superset/sql_parse.py

```diff
@@ -111,10 +111,13 @@ def is_select(self) -> bool:
         return self._parsed[0].get_type() == "SELECT"
 
     def is_explain(self) -> bool:
-        # Parsing SQL statement for EXPLAIN and filtering out comments
-        for statement in self.stripped().upper().splitlines():
-            if statement.upper().startswith("EXPLAIN"):
-                return True
+        # Remove comments
+        statements_without_comments = [statement.strip(" \t\n;") for statement in self.stripped().upper().splitlines() if not statement.startswith("--")]
```

Comment:
`sqlparse.format` offers a `strip_comments`  option that will also take care of `/*this*/` and probably more.